work/worker.go

```diff
@@ -404,16 +404,16 @@ func (self *worker) wait(TxResendUseLegacy bool) {
 
 			// Update the block hash in all logs since it is now available and not when the
 			// receipt/log of individual transactions were created.
-			for _, r := range work.receipts {
-				for _, l := range r.Logs {
-					l.BlockHash = block.Hash()
-				}
-			}
 			work.stateMu.Lock()
 			for _, log := range work.state.Logs() {
 				log.BlockHash = block.Hash()
 			}
 
+			var logs []*types.Log
+			for _, r := range work.receipts {
+				logs = append(logs, r.Logs...)
```

Comment:
Don't we need to set `BlockHash` here?

Reply:
`work.receipts` and `work.state.Logs()` are identical. Thus, removed duplicated assignment